### PR TITLE
37879:  Need a UI changes for SEP issue

### PR DESCRIPTION
**Pull request by sudhirreddysama**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [37879](https://devops.dchbx.org/redmine/issues/37879) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-Disqueued |
| Development Story Points | n/a |
| Peer Review Story Points | n/a |
| Ticket Author | @sudhirreddysama |

### CODE REVIEW

File: app/controllers/concerns/sep_all.rb [hunk 1]
```diff
@@ -150,7 +150,7 @@ def createSep
     date_arr.push(Date.strptime(params[:option2_date], "%m/%d/%Y").to_s) if params[:option2_date].present?
     date_arr.push(Date.strptime(params[:option3_date], "%m/%d/%Y").to_s) if params[:option3_date].present?
     special_enrollment_period.optional_effective_on = date_arr if date_arr.length > 0
-    special_enrollment_period.market_kind = params.permit(:market_kind)[:market_kind] if params[:market_kind].present?
+    special_enrollment_period.market_kind = qle.market_kind == "shop" ? "shop" : "ivl"
```
Comment (saikumar9) [inline app/controllers/concerns/sep_all.rb]: Picks correct **market_kind** for QLE.

File: app/controllers/insured/families_controller.rb [hunk 2]
```diff
@@ -89,6 +89,7 @@ def record_sep
       special_enrollment_period.selected_effective_on = Date.strptime(params[:effective_on_date], "%m/%d/%Y") if params[:effective_on_date].present?
       special_enrollment_period.qualifying_life_event_kind = qle
       special_enrollment_period.qle_on = Date.strptime(params[:qle_date], "%m/%d/%Y")
+      special_enrollment_period.market_kind = qle.market_kind == "shop" ? "shop" : "ivl"
```
Comment (saikumar9) [inline app/controllers/insured/families_controller.rb]: Picks correct **market_kind** for SEP based on QLE.

File: app/controllers/insured/family_members_controller.rb [hunk 3]
```diff
@@ -41,6 +41,7 @@ def index
       special_enrollment_period.selected_effective_on = Date.strptime(params[:effective_on_date], "%m/%d/%Y") if params[:effective_on_date].present?
       special_enrollment_period.qualifying_life_event_kind = qle
       special_enrollment_period.qle_on = Date.strptime(params[:qle_date], "%m/%d/%Y")
+      special_enrollment_period.market_kind = qle.market_kind == "shop" ? "shop" : "ivl"
```
Comment (saikumar9) [inline app/controllers/insured/family_members_controller.rb]: Picks correct **market_kind** for SEP based on QLE.

File: app/data_migrations/update_sep_market_kind.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+require File.join(Rails.root, "lib/mongoid_migration_task")
```
Comment (saikumar9) [inline app/data_migrations/update_sep_market_kind.rb]: Migration to fix the bad SEPs.

File: spec/data_migrations/update_sep_market_kind_spec.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
+require "rails_helper"
```
Comment (saikumar9) [inline spec/data_migrations/update_sep_market_kind_spec.rb]: Good test coverage.

File: app/controllers/concerns/sep_all.rb (hunk 1 above)
Comment (antonioschaffert) [reply app/controllers/concerns/sep_all.rb]: Should this be "ivl" or "individual" ?

File: app/controllers/insured/families_controller.rb (hunk 2 above)
Comment (antonioschaffert) [reply app/controllers/insured/families_controller.rb]: Should this be "ivl" or "individual" ?

File: app/controllers/insured/family_members_controller.rb (hunk 3 above)
Comment (antonioschaffert) [reply app/controllers/insured/family_members_controller.rb]: Should this be "ivl" or "individual" ?

File: app/controllers/concerns/sep_all.rb (hunk 1 above)
Comment (saikumar9) [reply app/controllers/concerns/sep_all.rb]: It should be **ivl** and not **individual** as this is how we have been storing.

File: app/controllers/insured/families_controller.rb (hunk 2 above)
Comment (saikumar9) [reply app/controllers/insured/families_controller.rb]: MarketKind of SEP should be **ivl** and not **individual** as this is how we have been storing.

File: app/controllers/insured/family_members_controller.rb (hunk 3 above)
Comment (saikumar9) [reply app/controllers/insured/family_members_controller.rb]: MarketKind of _SEP_ should be **ivl** and not **individual** as this is how we have been storing.

**Pull request merged**